code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm

```diff
@@ -155,16 +164,25 @@
 
 /// resist when widow does
 /datum/ai_behavior/spiderling/proc/parent_resist()
+	SIGNAL_HANDLER
 	var/mob/living/carbon/xenomorph/spiderling/spiderling_parent = mob_parent
 	spiderling_parent.do_resist()
 
 /// rest and unrest when widow does
-/datum/ai_behavior/spiderling/proc/toggle_rest()
-	var/mob/living/carbon/xenomorph/spiderling/spiderling_parent = mob_parent
-	if(HAS_TRAIT(spiderling_parent, TRAIT_FLOORED))
-		spiderling_parent.set_resting(FALSE)
-	else
-		spiderling_parent.set_resting(TRUE)
+/datum/ai_behavior/spiderling/proc/start_resting(mob/source)
+	SIGNAL_HANDLER
+	var/mob/living/living = mob_parent
+	living.set_resting(TRUE)
+
+/datum/ai_behavior/spiderling/proc/stop_resting(mob/source)
+	SIGNAL_HANDLER
+	var/mob/living/living = mob_parent
+	living.set_resting(FALSE)
+	source.unbuckle_all_mobs()
+
+/datum/ai_behavior/spiderling/proc/do_jump()
+	SIGNAL_HANDLER
+	SEND_SIGNAL(mob_parent, COMSIG_KB_LIVING_JUMP)
```

Comment:
Really not a fan of manually calling a keybind signal. Surely there's a better way. Also you didn't doc any of the procs.

Reply:
I'll doc the new procs, should I instead use COMSIG_ELEMENT_JUMP_STARTED? Thought it'd be better if the jumping was always going to happen at the same time, as it's a very timing-sensitive action